signac/contrib/project.py

```diff
@@ -2154,9 +2167,37 @@ def init_project(name, root=None, workspace=None, make_dir=True):
         configuration.
 
     """
-    return Project.init_project(
-        name=name, root=root, workspace=workspace, make_dir=make_dir
-    )
+    if args:
+        if len(args) == 1:
+            warnings.warn(
+                "Project names were removed in signac 2.0. If you intended to call "
+                "`init_project` with a root directory, please provided it as a keyword "
+                f"argument: `init_project(root={args[0]})`. If your project name "
+                "contains important information, consider storing it in the project document "
+                "instead.",
+                FutureWarning,
+            )
+        else:
+            # Match the usual error from misusing keyword-only args.
+            num_args = len(args)
+            raise TypeError(
+                f"init_project() takes 0 positional arguments but {num_args} "
+                "{'was' if num_args == 1 else 'were'} given"
+            )
+    if kwargs:
+        if len(kwargs) == 1 and "name" in kwargs:
+            warnings.warn(
+                "Project names were removed in signac 2.0. If your project name contains "
+                "important information, consider storing it in the project document instead.",
+                FutureWarning,
+            )
+        else:
+            # Match the usual error from extra keyword args.
+            raise TypeError(
+                f"init_project() got an unexpected keyword argument '{next(iter(kwargs))}'"
+            )
+
+    return Project.init_project(root=root, workspace=workspace, make_dir=make_dir)
```

Comment:
Just pass through `*args, **kwargs` here -- don't re-implement the error checking.